File: 203/cg_203_easy.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode tmpHead = new ListNode(-1);

        tmpHead.next = head;

        ListNode tmpNode = tmpHead;

        while (tmpNode.next != null) {
            while (tmpNode.next.val == val) {
                tmpNode.next = tmpNode.next.next;

                if (tmpNode.next == null) {
                    return tmpHead.next;
                }
            }

            tmpNode = tmpNode.next;
        }

        return tmpHead.next;
    }
}

File: 203/cg_203_easy.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        anchor = ListNode(-1)
        anchor.next = head
        node = anchor

        while node != None:
          if node.next != None and node.next.val == val:
            node.next = node.next.next
            continue

          node = node.next

        return anchor.next
